lib/network-layer/src/Cardano/Wallet/Network.hs

```diff
@@ -153,6 +153,9 @@ data NetworkLayer m block = NetworkLayer
     , stakeDistribution
         :: Coin -- Stake to consider for rewards
         -> m StakePoolsSummary
+    , getUTxOByTxIn
+        :: Set Write.TxIn
+        -> m (MaybeInRecentEra Write.UTxO)
```

Comment:
I think that we want

```suggestion
    , getUTxOByTxIn
        :: Set Read.TxIn
        -> m (Read.EraValue Read.UTxO)
```

at some point here — the function simply returns the `UTxO` in whatever `era` the node is currently in. But we can adjust that later, your version is fine for now.